### ISSUE TITLE
Can't edit and save previously saved sessions

### ISSUE THREAD

**Issue by cameroncooke** [ProxymanApp/Proxyman#935]
### Proxyman version? (Ex. Proxyman 1.4.3)

2.29.0 (22900)

### macOS Version? (Ex. mac 10.14)

12.0 Beta (21A5284e)

### Steps to reproduce 

1. Open Proxyman
2. Record some traffic
3. Save the session to disk and name it "A"
4. Close all the Proxyman tabs and windows
5. Quit Proxyman using Cmd+Q
6. Open the file named "A.proxymansession"
7. Edit the recorded history by deleting some requests
8. Attempt to save changes by tapping Cmd+S
9. Observe beep sound
10. Go to File -> Save Session
11. Give the session a new name "A1"
12. Close all the Proxyman tabs and windows
13. Quit Proxyman using Cmd+Q
14. Open the file named "A1.proxymansession"
15. Observe that the saved session isn't the one that was being viewed and edited and is in fact an entirely new recording 

### Expected behavior

Saved sessions should act like documents, opening viewing, and editing and saving should be possible. When the only tab open is a saved session Proxyman shouldn't be recording network traffic. 

The current behaviour is unintuitive and it doesn't feel like the tab behaviour has really been thought through.

**Comment by NghiaTranUIT** [comment]
> Saved sessions should act like documents, opening viewing, and editing and saving should be possible. 

It's a big obstacle that Proxyman is facing. We implement the document-based app incorrectly. Therefore, when we open a Session file, it doesn't support Auto-Save, or all native features of the Document-based app could benefit.

In order to fix it, we have to refactor how we implement this feature.

> When I record in Proxyman it records more than just the app or domain I care about, when I save a session it saves all traffic, not just the filtered view I'm seeing in the interface.

I highly recommend exporting as a ProxymanLog (Select your requests -> Right Click -> Export -> ProxymanLog), because you can select what domains/apps you would like to export (Not the entire session). You can also export entire Domain Node too (on the left panel).

-------------------------

I will bump this problem to the team and see how we can tackle it. Ideally, it should be the same as you mention 👍 😄

**Comment by cameroncooke** [comment]
Yeah, I think the document-based app design needs some work ;-) but it's all good. Great work on Proxyman I'm love it, hence all the feedback!
